### PR TITLE
Fix scan without otp prog

### PR DESCRIPTION
**Pull request by D-Triveni**
Fixes https://devzone.nordicsemi.com/f/nordic-q-a/103719/

### CODE REVIEW

File: samples/wifi/scan/src/main.c [hunk 1]
```diff
@@ -322,16 +322,22 @@ int main(void)
 	if (!is_mac_addr_set(net_if_get_default())) {
 		struct net_if *iface = net_if_get_default();
 		int ret;
+		struct ethernet_req_params params;
 
 		/* Set a local MAC address with Nordic OUI */
 		ret = net_if_down(iface);
-		if (ret) {
+		if (ret == -EALREADY) {
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:329]: This is fine, but you can only do below no need to check.

```
if (!net_if_is_up(iface)) {
   net_if_down(iface)
}
```

File: samples/wifi/scan/src/main.c [hunk 2]
```diff
@@ -332,8 +333,11 @@ int main(void)
 			return ret;
 		}
 
+		memcpy(params.mac_address.addr, CONFIG_WIFI_MAC_ADDRESS,
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:336]: This is wrong. `CONFIG_WIFI_MAC_ADDRESS` is a colon de-limited string, and `params.mac_address.addr` is a byte array.

You need to use a helper (something like reverse of `net_sprint_ll_addr_buf`) to do this translation.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/src/main.c [hunk 3]
```diff
@@ -332,8 +333,11 @@ int main(void)
 			}
 		}
 
+		net_bytes_from_str(params.mac_address.addr, sizeof(CONFIG_WIFI_MAC_ADDRESS),
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:336]: Please check the return value, I would also do some sanity checks on `CONFIG_WIFI_MAC_ADDRESS`

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/src/main.c [hunk 4]
```diff
@@ -322,16 +322,26 @@ int main(void)
...
+		if (net_if_is_up(iface)) {
+			ret = net_if_down(iface);
+			if (ret) {
+				LOG_ERR("Cannot bring down iface (%d)", ret);
+				return ret;
+			}
+		}
+
+		ret = net_bytes_from_str(params.mac_address.addr, sizeof(CONFIG_WIFI_MAC_ADDRESS),
+					CONFIG_WIFI_MAC_ADDRESS);
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:337]: alignmnt looks bit off

File: samples/wifi/scan/src/main.c [hunk 5]
```diff
@@ -322,16 +322,26 @@ int main(void)
...
 		if (ret) {
-			LOG_ERR("Cannot bring down iface (%d)", ret);
+			LOG_ERR("Failed to parse MAC address: %s", CONFIG_WIFI_MAC_ADDRESS);
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:339]: Please log the return value

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/src/main.c [hunk 6]
```diff
@@ -332,8 +333,16 @@ int main(void)
 			}
 		}
 
+		ret = net_bytes_from_str(params.mac_address.addr, sizeof(CONFIG_WIFI_MAC_ADDRESS),
+					 CONFIG_WIFI_MAC_ADDRESS);
+		if (ret) {
+			LOG_ERR("Failed to parse MAC address: %s (%d)",
+						CONFIG_WIFI_MAC_ADDRESS, ret);
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:340]: Looks like you are still using tabs, typically we align to open brace with tabs + spaces but without using too many tabs

**Review by krish2718 — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]
lgtm

**Pull request merged**